python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -348,7 +348,15 @@ def build_config(self) -> Optional[DgRawBuildConfig]:
             return None
 
         with open(build_yaml_path) as f:
-            return yaml.safe_load(f)
+            build_config_dict = yaml.safe_load(f)
+            build_directory = build_config_dict.get("directory")
+            if build_directory:
+                build_directory_path = Path(build_directory)
+                if not build_directory_path.is_absolute():
+                    build_directory_path = (build_yaml_path.parent / build_directory_path).resolve()
```

Comment:
This seems like the right call, and the least confusing option, though I do worry it is still somewhat confusing